.github/workflows/lint.yaml

```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    name: "Lint"
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node.js
+        uses: actions/setup-node@master
+        with:
+          node-version: 14.4.0
+
+      - uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Install dependencies
+        if: steps.cache-deps.outputs.cache-hit != 'true'
+        run: yarn install --nonInteractive --frozen-lockfile
+
+      - name: Lint code
+        run: yarn run lint:code:ci
```

Comment:
Will there be a case where we don't lint code? If not, can we change to just `lint:ci`?

Reply:
There could also be lint:types which run tsc as a type checker. If you would like we can have lint:ci to just be eslint and have
something like `check:types` for type checking?

I'm gonna tackle type checking in separate PR. Issue is that you have excluded test files from tsconfig. I assume because you don't wan't them in your dist/build files but that also means they never get's type checked. Some common solution is to have tsconfig.json which includes everything and tsconfig.build.json which extends from tsconfig.json but excludes tests.

Reply:
Okay that makes sense. Yes, having a tsconfig.build.json or tsconfig.dist.json sounds like a good solution here.